src/Microsoft.Health.Fhir.Api/Controllers/FhirController.cs

```diff
@@ -402,6 +408,28 @@ void AddItemsIfNotNull(IEnumerable<KeyValuePair<string, StringValues>> source)
             }
         }
 
+        /// <summary>
+        /// Reads the specified resource.
+        /// </summary>
+        /// <param name="compartmentType">The compartment type.</param>
+        /// <param name="id">The identifier.</param>
+        /// <param name="type">The resource type.</param>
+        [HttpGet]
+        [Route(KnownRoutes.CompartmentTypeByResourceType, Name = RouteNames.ReadCompartmentByResourceType)]
+        [AuditEventSubType(AuditEventSubType.ReadCompartmentByResourceType)]
+        public async Task<IActionResult> ReadCompartmentByResourceType(string compartmentType, string id, string type)
```

Comment:
I think compartment falls under search not read since it returns multiple documents.